### PR TITLE
adding directionality to parameters.

### PR DESCRIPTION
**Pull request by vsoch**
This can currently be import/export but I expect this will change. I'm also wondering if it's the case that null/undefined is a primitive type, how will that show up as a type string?

Signed-off-by: vsoch <vsoch@users.noreply.github.com>

### CODE REVIEW

File: source/detail/parser/x86_64.cpp
Comment (hainest) [inline source/detail/parser/x86_64.cpp:137]: The algorithm here is a bit more complicated because we need to handle typedefs and multiple indirections. The code doesn't need to change completely-- just adding more checks.

File: source/detail/parser/x86_64.cpp
Comment (hainest) [inline source/detail/parser/x86_64.cpp:142]: We can use Dyninst to do this for us. Refactoring it into its own function would be good since we'll need it in several spots, as well.

File: source/detail/parser/x86_64.cpp
Comment (hainest) [inline source/detail/parser/x86_64.cpp:152]: I believe from here we want the default to be 'import' since we're handling symbols which are function definitions.

**Review by hainest — CHANGES_REQUESTED** [review]

File: source/detail/parser/x86_64.cpp
Comment (vsoch) [reply source/detail/parser/x86_64.cpp:139]: Do you want to talk about it, or just take a shot?

File: source/detail/parser/x86_64.cpp
Comment (vsoch) [reply source/detail/parser/x86_64.cpp:144]: I tried doing this locally but it was the wrong type, so I'll need your help here.

File: source/detail/parser/x86_64.cpp
Comment (vsoch) [reply source/detail/parser/x86_64.cpp]: okay, updated.

File: source/detail/parser/x86_64.cpp
Comment (vsoch) [reply source/detail/parser/x86_64.cpp]: done!

**Comment by hainest** [comment]
Derp. I wrote some bugs in this. It doesn't handle `typedef int* Foo` correctly (it doesn't realize there is a pointer there). Let me fix them and I'll send you a diff.

**Comment by hainest** [comment]
Github won't let me upload a .patch file, so I stuck an extra suffix on it.

[param_typedef.patch.log](https://github.com/buildsi/Smeagle/files/6634875/param_typedef.patch.log)

**Comment by vsoch** [comment]
@hainest why not just pull the branch and add it? I have no problem with that!

**Comment by vsoch** [comment]
@hainest does this look good? Do you want to chat about what I can do next?

**Comment by hainest** [comment]
Ah, I wasn't sure if I had write access to this repo. Good to know! It looks good to me. Do you have some time tomorrow to chat about where to go next?

**Comment by vsoch** [comment]
I do! I'll probably run around 11:30am Mountain so if you ping me before that I should be around. Should we merge here?

**Comment by hainest** [comment]
Let's try for 11AM MST. I'm good to merge, if you are.

**Pull request merged**
